functionary/ui/views/functions.py

```diff
@@ -17,18 +17,23 @@
 class FunctionListView(PermissionedEnvironmentListView):
     model = Function
     environment_through_field = "package"
+    queryset = Function.objects.select_related("package").all()
     order_by_fields = ["package__name", "name"]
 
 
 class FunctionDetailView(PermissionedEnvironmentDetailView):
     model = Function
     environment_through_field = "package"
 
+    def get_queryset(self):
+        return super().get_queryset().select_related("package", "package__environment")
+
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        env = Environment.objects.get(id=self.request.session.get("environment_id"))
+        the_function = self.get_object()
```

Comment:
`function` is not a reserved word or builtin, so this can just be `function` rather than `the_function`.